appserver/jdbc/jdbc-config/src/main/java/org/glassfish/jdbc/config/validators/JdbcConnectionPoolValidator.java

```diff
@@ -41,204 +41,234 @@
+import org.glassfish.jdbc.config.JdbcConnectionPool;
 
 /**
  * Implementation for Connection Pool validation.
- * Following validations are done :
- * - Validation of datasource/driver classnames when resource type is not null 
- * - Max pool size to be always higher than steady pool size
- * - Check if statement wrapping is on when certain features are enabled.
- * 
+ *
+ * Following validations are done:
+ * <ul>
+ * <li>Validation of datasource/driver classnames when resource type is not null
+ * <li>Max pool size to be always higher than steady pool size
+ * <li>Check if statement wrapping is on when certain features are enabled.
+ * </lu>
+ *
  * @author Shalini M
  */
-public class JdbcConnectionPoolValidator
-    implements ConstraintValidator<JdbcConnectionPoolConstraint, ResourcePool> {
-    
+public class JdbcConnectionPoolValidator implements ConstraintValidator<JdbcConnectionPoolConstraint, ResourcePool> {
+
     protected ConnectionPoolErrorMessages poolFaults;
-    
+
+
+    @Override
     public void initialize(final JdbcConnectionPoolConstraint constraint) {
         this.poolFaults = constraint.value();
     }
 
-    public String getParsedVariable(String variableToRetrieve) {
 
-        String[] variableReference = variableToRetrieve.split("=");
+    @Override
+    public boolean isValid(final ResourcePool pool, final ConstraintValidatorContext context) {
+        if (!JdbcConnectionPool.class.isInstance(pool)) {
+            return true;
+        }
+        // original code allowed this. Why?
+        if (this.poolFaults == null) {
+            return true;
+        }
+        final JdbcConnectionPool jdbcPool = (JdbcConnectionPool) pool;
+        switch (poolFaults) {
+            case POOL_SIZE_STEADY:
+                return validateSteadyPoolSize(jdbcPool);
+            case POOL_SIZE_MAX:
+                return validateMaxPoolSize(jdbcPool);
+            case STMT_WRAPPING_DISABLED:
+                return validateWrapping(jdbcPool);
+            case TABLE_NAME_MANDATORY:
+                return validateTableConnectionValidation(jdbcPool);
+            case CUSTOM_VALIDATION_CLASS_NAME_MANDATORY:
+                return validateClassConnectionValidation(jdbcPool);
+            case RES_TYPE_MANDATORY:
+                return validateResourceType(jdbcPool);
+            default:
+                return true;
+        }
+    }
 
-        if (variableReference.length == 1) {
-            //We got a system variable as no split occured
-            return System.getProperty(variableReference[0]);
+
+    private boolean validateSteadyPoolSize(final JdbcConnectionPool jdbcPool) {
+        final Integer steadyPoolSize = getSteadyPoolSize(jdbcPool);
+        return steadyPoolSize != null && steadyPoolSize >= 0;
+    }
+
+
+    private boolean validateMaxPoolSize(final JdbcConnectionPool jdbcPool) {
+        final Integer steadyPoolSize = getSteadyPoolSize(jdbcPool);
```

Comment:
final for local variables are overhead

Reply:
Additional info to the response I wrote to @Pandrex247, surprising even myself: https://stackoverflow.com/a/21485260/375449
It seems finals have much bigger positive effect then I thought ... (I thought on JVMs after JDK5 or 6 it is nearly same, but it is not. We should use final variables much more than we do ...)

Reply:
(this is rather argumentation foul, but it is still interesting result)
https://dzone.com/articles/all-about-finals

Reply:
>We should use final variables much more than we do ...)

Well, then we should have some code rules and update all code in the repository to be uniform. Now we start creating a mixture which is 'ugly'.

And it is misleading, final doesn't mean the variable can't be changed, just the reference (so in most cases the 'content' of the object can be updated - unless it is created as immutable).

Reply:
There are no standards in Payara/Glassfish code, but I agree we can standardize it (then we must use tools to enforce that). We can discuss it privately on monday, this is not right place for that.
The `final` keyword was in Java always about references, not internal states of objects.